### PR TITLE
[ waiting on #12594 ] Membership code cleanup

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Refactors a function in Membership & Membership renewal to be shared

Before
----------------------------------------
Code less shared

After
----------------------------------------
Code more shared

Technical Details
----------------------------------------
@agilewarealok I wound up digging in this code & did this minor refactor - perhaps you could review.

I think I'll separate out the Sales Tax trait to another PR though

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

### CODE REVIEW

File: CRM/Member/Form.php [hunk 1]
```diff
@@ -465,4 +472,40 @@ public function testSubmit($formValues) {
...
+   *
+   * @param array $values
+   * @param string $totalAmount
+   * @param string $taxAmount
+   *   (only used from membership renewal)
+   *
+   * @return array
+   */
+  protected function getMembershipInfoArray($values, $totalAmount, $taxAmount = NULL) {
+    // @todo - we really should assign the tax rate for the financial_type_id to the template.
```
Comment (eileenmcnaughton) [inline CRM/Member/Form.php]: opps this todo can go ... I did & later I found using it improved the tpl - but thinking to hold this & rebase when the trait pr is merged

**Comment by mattwire** [comment]
@eileenmcnaughton Could you flag as waiting on #12594 before review

**Comment by eileenmcnaughton** [comment]
@mattwire - I've updated per your comments - if I can't get   #12594 reviewed moderately soon I'll abandon this as I'm more vested in some of my other PRs

**Comment by eileenmcnaughton** [comment]
I went to rebase this & hit a couple of issues - I could re-do the refactor but it will need a reviewer so I'm gonna leave it unless someone is interested in reviewing

**Pull request closed without merging**